sdk/storage/azure-storage-common/CMakeLists.txt

```diff
@@ -73,10 +73,9 @@ target_include_directories(
     PUBLIC
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
       $<INSTALL_INTERFACE:include>
-      ${azure-core-cpp_INCLUDE_DIRS}
 )
 
-target_include_directories(azure-storage-common INTERFACE ${LIBXML2_INCLUDE_DIRS})
+target_include_directories(azure-storage-common PUBLIC ${LIBXML2_INCLUDE_DIRS})
```

Comment:
Hi @antkmsft , this is a changed made in PR https://github.com/Azure/azure-sdk-for-cpp/pull/1013. Can you help us understand why `INTERFACE` is perferred over `PUBLIC`?

We found this caused a build error of `libxml2` not being found. It can build if dependencies are installed via `vcpkg` just because `libxml2` headers happen to be in the same directory as other dependencies.

Reply:
It is the other way around - when something is PUBLIC, it carries the dependencies with it. From what I can see, switching to `PUBLIC` from `INTERFACE` is ok from the perspective of building vcpkg. You don't want to do that if you don't use libxml2 in your public headers (#1009), but then it should be `PRIVATE`.
Right now, it is not this library that fails compilation, it is probably the application that consumes the SDK fails to build, in which case, if the library is consumed as `INTERFACE`, it should be doing find_package() in it's own cmakefile. 

In short, it is fine to switch it from INTERFACE to PUBLIC, but if #1009 gets fixed, it should become PRIVATE.

Reply:
I guess we have different understanding for `INTERFACE` and `PUBLIC`.

Be it as it may, it doesn't compile if it was `INTERFACE`. I also agree after #1009 is fixed, this should be changed to `PRIVATE`.

Reply:
> Right now, it is not this library that fails compilation, it is probably the application that consumes the SDK fails to build

I added the error message, it failed when building `azure-storage-common`.

Reply:
I am surprised, because CI builds successfully, and it builds successfully in the isolation as well (the screenshots in this PR: https://github.com/Azure/azure-sdk-for-cpp/pull/1013)

Reply:
as I already said, CI succeeded because we only test the scenario of installing dependencies via vcpkg. In this case, libxml headers happen to be in the same directory as other dependencies.

Reply:
I am fine if you switch INERFACE to PUBLIC. I don't think you should be doing that, but it won't break vcpkg build, the rest is up to you. The ideal is PRIVATE, after all. My only objection is the removal of `target_include_dirs(${azure-*_INCLUDE_DIRS})`.

Reply:
I'll resolve this conversation.